v1/topdown/eval.go

```diff
@@ -2084,7 +2101,13 @@ func (e evalFunc) evalValue(iter unifyIterator, argCount int, findOne bool) erro
 		}
 
 		if e.ir.Default != nil && prev == nil {
-			_, err := e.evalOneRule(iter, e.ir.Default, cacheKey, prev, findOne)
+			copy(args, e.ir.Default.Head.Args)
+			if len(args) == len(e.ir.Default.Head.Args)+1 {
+				args[len(args)-1] = e.ir.Default.Head.Value
+			}
```

Comment:
We're jumping through this hoop three times: smells like a helper function ;).

Reply:
Possibly, but it's _slightly_ different each time... 🙂 I'll see if I can simplify this in my next PR. Thanks!